lib/mixlib/shellout/windows.rb

```diff
@@ -122,7 +122,7 @@ def run_command
                     wmi = WmiLite::Wmi.new
                     kill_process_tree(process.process_id, wmi, logger)
                     Process.kill(:KILL, process.process_id)
-                  rescue
+                  rescue SystemCallError
```

Comment:
Are there one or two `Errno` exceptions that we're really wanting to catch? Are we catching `SystemCallError` to intentionally catch everything from Errno?

```
irb(main):010:0> Errno::ENOENT.ancestors
=> [Errno::ENOENT, SystemCallError, StandardError, Exception, Object, Kernel, BasicObject]
irb(main):011:0> Errno.constants
=> [:NOERROR, :EPERM, :ENOENT, :ESRCH, :EINTR, :EIO, :ENXIO, :E2BIG, :ENOEXEC, :EBADF, :ECHILD, :EAGAIN, :ENOMEM, :EACCES, :EFAULT, :ENOTBLK, :EBUSY, :EEXIST, :EXDEV, :ENODEV, :ENOTDIR, :EISDIR, :EINVAL, :ENFILE, :EMFILE, :ENOTTY, :ETXTBSY, :EFBIG, :ENOSPC, :ESPIPE, :EROFS, :EMLINK, :EPIPE, :EDOM, :ERANGE, :EDEADLK, :ENAMETOOLONG, :ENOLCK, :ENOSYS, :ENOTEMPTY, :ELOOP, :EWOULDBLOCK, :ENOMSG, :EIDRM, :ECHRNG, :EL2NSYNC, :EL3HLT, :EL3RST, :ELNRNG, :EUNATCH, :ENOCSI, :EL2HLT, :EBADE, :EBADR, :EXFULL, :ENOANO, :EBADRQC, :EBADSLT, :EDEADLOCK, :EBFONT, :ENOSTR, :ENODATA, :ETIME, :ENOSR, :ENONET, :ENOPKG, :EREMOTE, :ENOLINK, :EADV, :ESRMNT, :ECOMM, :EPROTO, :EMULTIHOP, :EDOTDOT, :EBADMSG, :EOVERFLOW, :ENOTUNIQ, :EBADFD, :EREMCHG, :ELIBACC, :ELIBBAD, :ELIBSCN, :ELIBMAX, :ELIBEXEC, :EILSEQ, :ERESTART, :ESTRPIPE, :EUSERS, :ENOTSOCK, :EDESTADDRREQ, :EMSGSIZE, :EPROTOTYPE, :ENOPROTOOPT, :EPROTONOSUPPORT, :ESOCKTNOSUPPORT, :EOPNOTSUPP, :EPFNOSUPPORT, :EAFNOSUPPORT, :EADDRINUSE, :EADDRNOTAVAIL, :ENETDOWN, :ENETUNREACH, :ENETRESET, :ECONNABORTED, :ECONNRESET, :ENOBUFS, :EISCONN, :ENOTCONN, :ESHUTDOWN, :ETOOMANYREFS, :ETIMEDOUT, :ECONNREFUSED, :EHOSTDOWN, :EHOSTUNREACH, :EALREADY, :EINPROGRESS, :ESTALE, :EUCLEAN, :ENOTNAM, :ENAVAIL, :EISNAM, :EREMOTEIO, :EDQUOT, :ECANCELED, :EKEYEXPIRED, :EKEYREJECTED, :EKEYREVOKED, :EMEDIUMTYPE, :ENOKEY, :ENOMEDIUM, :ENOTRECOVERABLE, :EOWNERDEAD, :ERFKILL, :EAUTH, :EBADRPC, :EDOOFUS, :EFTYPE, :ENEEDAUTH, :ENOATTR, :ENOTSUP, :EPROCLIM, :EPROCUNAVAIL, :EPROGMISMATCH, :EPROGUNAVAIL, :ERPCMISMATCH, :EIPSEC, :EHWPOISON, :ECAPMODE, :ENOTCAPABLE]
```

Reply:
Hi @btm, We need to catch `Errno::EIO, SystemCallError` as per master branch here https://github.com/chef/mixlib-shellout/pull/157/files#diff-a49639dc02ce0578d8000ee0aecf5210L125 but that was causing rubocop errors so removed to pass the rubocop.

Reply:
so safest thing to do here is probably just to replace `rescue Errno::EIO, SystemCallError` with `rescue SystemCallError` which will have identical behavior to the previous code and should keep rubocop happy

Reply:
Yep, the cop is poorly worded in some ways but all it is complaining about is that you have a redundant or unreachable (in this case the former) rescue. In the future, you can always look up the name of the failing cop to get more info about it.

Reply:
All right there is a way to fix that rubocop warning `Do not shadow rescued exceptions`, 
Like:
```
# Call required exceptions in constant
SERVER_EXCEPTIONS = [Errno::EIO, SystemCallError]
# Then in code call constant in rescue block
begin
    // your code
rescue SERVER_EXCEPTIONS
   // your exceptions
end
```
Please check if that works for you and let me know your thought about that.

Reply:
Okay, let's rewind. `Errno::EIO` is a subclass of `SystemCallError`. That means rescuing both exceptions is redundant. Which is why Rubocop triggers here. It would be like checking `if x == 1 && x < 10`. You don't need to check both because one is strictly part of the other. So as Lamont said, all you need is `rescue SystemCallError`.